smithy-model/src/test/java/software/amazon/smithy/model/loader/ModelAssemblerTest.java

```diff
@@ -88,6 +92,26 @@ public void addsExplicitShapes() {
         assertThat(result.unwrap().getShape(ShapeId.from("ns.foo#Bar")), is(Optional.of(shape)));
     }
 
+    @Test
+    public void addsExplicitTraits() {
+        StringShape shape = StringShape.builder().id("ns.foo#Bar").build();
+
+        Node node = Node.fromStrings("validator");
+        TraitFactory provider = TraitFactory.createServiceFactory();
+        Optional<Trait> trait = provider.createTrait(
+                ShapeId.from("smithy.api#suppress"), shape.toShapeId(), node);
+
+        SuppressTrait suppress = (SuppressTrait) trait.get();
+        ValidatedResult<Model> result = new ModelAssembler()
+                .addShape(shape)
+                .addTrait(shape.toShapeId(), suppress)
+                .assemble();
+
+        assertThat(result.getValidationEvents(), empty());
+        Shape resultShape = result.unwrap().getShape(ShapeId.from("ns.foo#Bar")).get();
+        assertTrue(resultShape.findTrait("smithy.api#suppress").isPresent());
```

Comment:
Then here you could test both for the concrete trait class, no? (i.e., `shape.getTrait(SuppressTrait.class)`)